doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+
+``domain``: `Set`
+    The domain of this functional, i.e., the set of elements to which this functional can be applied.
+
+Moreover, there are several optional parameters associated with a functional. These are
+
+``linear``: `bool`, optional
+    If `True`, the functional is considered as linear.
+    Default: ``False``
+``smooth``: `bool`, optional
+    If `True`, assume that the functional is continuously differentiable.
+    Default: ``False``
+``concave``: `bool`, optional
+    If `True`, assume that the functional is concave.
+    Default: ``False``
+``convex``: `bool`, optional
+    If `True`, assume that the functional is convex.
+    Default: ``False``
+``grad_lipschitz``: `float`, optional
+    The Lipschitz constant of the gradient.
+    Default: infinity
+
+A functional also has a number of optional methods and properties associated with it.
+The default value of these are all to raise a `NotImplemetedError`.
+The properties are:
+
+ * ``gradient``. This returns the gradient operator of the functional, i.e., the operator that corresponds to the mapping
+
+   .. math::
+
+      x \to \nabla f(x),
+
+   where :math:`\nabla f(x)` is the element used to evaluate the (directional) derivative in a direction :math:`d` by :math:`\langle \nabla f(x), d \rangle`.
+
+ * ``conjugate_functional``. This is the convex conjugate functional, also known as the Legendre transform or Fenchel conjugate.
+   It is defined as
+
+   .. math::
+
+      f^*(x^*) = \sup_{x \in X} \{ \langle x^*,x \rangle - f(x)  \},
+
+   where the element :math:`x^*` lives in the space :math:`X^*`, the (continuous/normed) dual space of :math:`X`, i.e., the space of all continuous linear functionals defined on :math:`X`.
+   Moreover, the notation :math:`\langle x^*,x \rangle` is the "dual pairing" between :math:`X` and :math:`X^*`, i.e., the evaluation of the linear functional :math:`x^*` in the point :math:`x`.
+   However, Hilbert spaces are self-dual, meaning :math:`X^* = X`, and :math:`\langle x^*,x \rangle` is then the inner product.
+
+The optional method is:
+
+ * ``proximal(sigma)``. This returns the proximal operator of the functional, where ``sigma`` is a nonnegative step-size like parameter.
+   The proximal operator is defined as
+
+   .. math::
+
+      \text{prox}_{\sigma f}(x) = \text{arg min}_{y \in X} \{ f(y) + \frac{1}{2\sigma} \|y - x\|_2^2 \}. 
+
+The `Functional` class also contains two default implementations of two help functions:
+
+ * ``derivative(point)``. Given an implementation of the gradient, this method return the (directional) derivative operator in ``point``.
+   This is the linear operator
+
+   .. math::
+
+      x \to \langle x, \nabla f(point) \rangle,
+
+   where :math:`\nabla f(point)` is the gradient of the functional in the point :math:`point`.
+
+ * ``translate(shift)``. Give a functional :math:`f(.)`, this method creates the functional :math:`f(. - shift)`
+
+
+Functional arithmetics
```

Comment:
Not sure if "arithmetics" is correct, I get wiggly lines when writing it here. Perhaps use "arithmetic" or "calculus" instead?